src/Lorentz/Contracts/BaseDAO/TZIP16Metadata.hs

```diff
@@ -30,17 +38,86 @@ knownBaseDAOMetadata = mconcat
   , views baseDAOViews
   ]
 
+------------------------------------------------------------------------
+-- Off-chain views
+------------------------------------------------------------------------
+
+type DaoView = View (ToT MetadataStorage)
+
 baseDAOViews :: [View $ ToT MetadataStorage]
 baseDAOViews =
-  [ View
-    { vName = "GetCounter"
-    , vDescription = Just
-        "Returns the next counter value with which a permit should be created."
-    , vPure = Just True
-    , vImplementations =
-        [ VIMichelsonStorageView $
-            mkSimpleMichelsonStorageView @MetadataStorage . WithoutParam $
-              stToField #sPermitsCounter
-        ]
-    }
+  [ getBalanceView
+  , allTokensView
+  , isOperatorView
+    -- total_supply is skipped since the necessary info is not tracked
+    -- add token_metadata? - TODO for next commits
+
+  , permitsCounterView
   ]
+
+-- FA2
+------------------------------------------------------------------------
+
+getBalanceView :: DaoView
+getBalanceView = View
+  { vName = "get_balance"
+  , vDescription = Just
+      "Get balance of an address according to TZIP-12."
+  , vPure = Just True
+  , vImplementations =
+      [ VIMichelsonStorageView $
+          mkMichelsonStorageView @MetadataStorage @Natural Nothing [] $
+            WithParam @FA2.BalanceRequestItem $ do
+              forcedCoerce_
```

Comment:
What is coerced here and why can't it be done safer?

Reply:
`FA2.BalanceRequestItem` to `LedgerKey`. Yeah, apparently I better write this via getters.